datumaro/plugins/data_formats/datumaro/exporter.py

```diff
@@ -64,68 +65,89 @@ def items(self):
     def is_empty(self):
         return not self.items
 
-    def add_item(self, item: DatasetItem):
-        annotations = []
-        item_desc = {
-            "id": item.id,
-            "annotations": annotations,
-        }
-
-        if item.attributes:
-            item_desc["attr"] = item.attributes
-
-        if isinstance(item.media, Image):
+    @contextmanager
+    def save_media(self, item: DatasetItem):
+        """Implicitly change the media path and save it if save_media=True.
+        When done, revert it's path as before.
+        """
+        if item.media is None:
+            yield
+        elif isinstance(item.media, Image):
             image = item.media_as(Image)
             path = image.path
+
             if self._context._save_media:
-                path = self._context._make_image_filename(item)
-                self._context._save_image(
-                    item, osp.join(self._context._images_dir, item.subset, path)
+                # Temporarily update image path and save it.
+                image._path = osp.join(
+                    self._context._images_dir, item.subset, self._context._make_image_filename(item)
                 )
+                self._context._save_image(item, image.path)
 
-            item_desc["image"] = {
-                "path": path,
-            }
-            if item.media.has_size:  # avoid occasional loading
-                item_desc["image"]["size"] = item.media.size
+            yield
+            image._path = path
         elif isinstance(item.media, PointCloud):
             pcd = item.media_as(PointCloud)
             path = pcd.path
+
             if self._context._save_media:
-                path = self._context._make_pcd_filename(item)
-                self._context._save_point_cloud(
-                    item, osp.join(self._context._pcd_dir, item.subset, path)
+                # Temporarily update pcd path and save it.
+                pcd._path = osp.join(
+                    self._context._pcd_dir, item.subset, self._context._make_pcd_filename(item)
                 )
+                self._context._save_point_cloud(item, pcd.path)
+
+                # Temporarily update pcd related images paths and save them.
+                for i, img in enumerate(sorted(pcd.extra_images, key=lambda v: v.path)):
+                    img.__path = img.path
```

Comment:
I am sorry that I was failed to find `__path`. Could you let me know?

Reply:
This variable is used to save the original path temporariliy.